ouroboros-consensus/test-storage/Test/Ouroboros/Storage/VolatileDB/Model.hs

```diff
@@ -121,6 +124,14 @@ getBlockModel err sl = do
     if not open then EH.throwError' err $ UserError ClosedDBError
     else return $ Map.lookup sl mp
 
+getHeaderModel :: forall m blockId. (MonadState (DBModel blockId) m, Ord blockId)
+               => ThrowCantCatch VolatileDBError m
+               -> blockId
+               -> m (Maybe (SlotNo, ByteString))
+getHeaderModel err sl = do
+    bs <- getBlockModel err sl
+    return $ second (BL.take 1 . BL.drop (fromIntegral headerOffset)) <$> bs
```

Comment:
In fact, why is this hardcoded to `TestBlock`? This means that the mock VolatileDB will always use 1 as header size, regardless the actual block.

The model should be parametric in the block, except maybe for some utilities in the tests (and things like `FileCorruption`).

An option is to use `Test.Ouroboros.Storage.TestBlock`, just like the ImmutableDB and the ChainDB.